Update User Guide based on in class review (#122)
AY2122S2-CS2103T-T09-2/tp: docs/UserGuide.md

---
### Adding an applicant: `add`

Allows you to add a new applicant to be tracked with the following details:
* Name
* Phone Number
* Email
* The job they've applied for
* Round of job application
* Particular skills they may have

<br>

Format:
```
add n/NAME p/PHONE_NUMBER e/EMAIL j/JOB r/ROUND [s/SKILLS]...
```
<div markdown="span" class="alert alert-primary">:bulb: **Tip:**

> A person can have any number of skills (including none at all)

Reviewer: Is it better to state it as an applicant?
```suggestion
An applicant can have any number of skills (including none at all)
```

Author: Good spot, thanks!

---
### Viewing a specific applicant : `view`

Allows you to view an overview of a specific applicant, specified by an applicant's full name (exact match)

<br>

Format:
```
view NAME
```
* The search is case-insensitive. e.g `hans` will match `Hans`
* Only the name is searched.
* Only full name will be matched e.g. `Han Lee` will not match `Han`
* Only exact full name with correct spacing will be matched e.g. `HanLee` will not match `Han Lee`

<br>

Example:
```
view Steve Jobs
```
Sample Output:
```
Name: Steve Jobs;
Phone: 99999999;
Email: stevejobs@apple.mail;
Job: iOS Developer;
Round: Hired;
Skills: Swift
```

[Back to top <img src="images/back-to-top-icon.png" width="25px" />](#table-of-contents)

---
### Searching a specific applicant : `search`

Allows you to view an overview of a specific applicant, specified by **either** an applicant's name, job, round or skills.

<br>

Format:
```
search [n/NAME]... [j/JOB]... [r/ROUND]... [s/SKILL]...
```
* The search is case-insensitive. e.g `hans` will match `Hans`
* First name/Last name will be matched e.g. `Han` will match `Han Lee`
* Only full words will be matched e.g. `Han` will not match `Hans`

> * Search will not work with multiple prefixes. e.g `search n/Alex j/Interview` will only yield results of `search n/Alex`

Reviewer: which version of search are we suppose to include?

Author: Yup I have updated UG to work with your latest PR.

<div markdown="span" class="alert alert-primary">:bulb: **Tip:**

> You can try searching for multiple names in the applicant list. Try `search n/Alex Bernice` to look up these 2 applicants!

Reviewer: is this better?
```suggestion
You can try searching for multiple names in the applicant list. Try `search n/Alex Bernice` to search for applicants with name 'Alex' or 'Bernice'!
```

Author: Thanks for the suggestion, will add it in.

<br>

Format:
```
flag INDEX
```

* Only valid indexes are flagged. e.g If there are only `4` applicants in the app but `5` is specified, then the intended action will not be carried out.
* Only positive indexes are flagged. e.g As we label our applicants starting from `1...`, an index of `-1` will not be tagged to an applicant.
* Flag acts like a toggle. To un-flag the applicant, you may simply re-type the same command.

<div markdown="span" class="alert alert-primary">:bulb: **Tip:**
Try flagging the first applicant by typing `flag 1`. You should see it at the top with a flag symbol. Then try un-flagging the same applicant with `flag 1`.
</div>

<br>

Example:
```
flag 1
```
Sample Output:
```
Flagged Applicant: David Lee; 
Phone: 91234567; 
Email: johndoe@example.com; 
Job: Risk Assessment Associate; 
Round: Stock Pitch Assessment; 
Skills: [Accounting][Equities][Cryptocurrency]
```

[Back to top <img src="images/back-to-top-icon.png" width="25px" />](#table-of-contents)

---
### Deleting an applicant : `delete`

> Allows you to delete a specific job applicant, specified by index.

Reviewer: seems like 'applicant' is used in other features, probably better to standardize it?
```suggestion
Allows you to delete a specific applicant, specified by index.
```

Author: Fixed!
